### PR TITLE
[Fix] The voice message problem

### PR DESCRIPTION
**Pull request by KaterinaWire**
### Issues
A voice message from the android device can not be played back.

### Solutions
Check if we have a file extension for the audio file, and add it if we did not receive it.

### CODE REVIEW

File: Source/Model/Message/ZMAssetClientMessage+FileMessageData.swift [hunk 1]
```diff
@@ -150,6 +159,15 @@ extension ZMAssetClientMessage: ZMFileMessageData {
         return temporaryFileURL
     }
     
+    private var fileExtension: String? {
```
Comment (billypchan) [inline Source/Model/Message/ZMAssetClientMessage+FileMessageData.swift]: I found there is a similar code in `FilePreviewGenerator.swift` (`URL.UTI()`), may be we can reuse it.

Comment (KaterinaWire) [reply Source/Model/Message/ZMAssetClientMessage+FileMessageData.swift]: But this code we have in UI project and I need to find a file extension based on mimeType

File: Source/Model/Message/ZMAssetClientMessage+FileMessageData.swift [hunk 2]
```diff
@@ -132,9 +133,17 @@ extension ZMAssetClientMessage: ZMFileMessageData {
...
+            let filename = filename,
+            let temporaryDirectoryURL = temporaryDirectoryURL else { return nil }
         
-        let temporaryFileURL = temporaryDirectoryURL.appendingPathComponent(filename)
+        var temporaryFileURL = temporaryDirectoryURL.appendingPathComponent(filename)
+        
+        if let fileExtension = fileExtension,
+            richAssetType == .audio,
+            temporaryFileURL.pathExtension != fileExtension {
+            temporaryFileURL = temporaryFileURL.appendingPathExtension(fileExtension)
```
Comment (typfel) [inline Source/Model/Message/ZMAssetClientMessage+FileMessageData.swift]: You can use the mutating version `appendPathExtension()` then you need to re-assign `temporaryFileURL`.

Comment (KaterinaWire) [reply Source/Model/Message/ZMAssetClientMessage+FileMessageData.swift]: Fixed

**Comment by KaterinaWire** [comment]
@zenkins release patch

**Pull request merged**